src/components/AppBrowser.js

```diff
@@ -45,25 +58,42 @@ export default {
         this.$router.push(app.main)
       }
     },
-    back() {
+    back () {
+      this.$router.push(PATHS.EMBEDDED_APP)
+      this.iframe = ''
       this.showIframe = false
     },
-    add() {
-      let url = prompt('URL');
-      if(url) {
+    add () {
+      let url = prompt('URL')
+      if (url) {
         this.$store.dispatch('addApp', url)
       }
-
     },
+    resolveUrl (url) {
+      if (url) {
+        let app = this.apps.find((app) => {
+          return app.main.indexOf(url) > -1
+        })
+        if (app) {
+          this.open(app)
+        } else {
+          this.$store.dispatch('addApp', url).then((app) => {
```

Comment:
Should be better to ask the user before adding an application required by the opened deep link.

Reply:
The currently displayed app and possible parameters to it should become properties of AppBrowser. If adding an app is necessary, should be decided by the store.

The router takes the information which app to use and with what parameters, Than tells the store to add the app if necessary and than routes to AppBrowser with the app's identifier and the parameters as properties.